openapi-spec/swagger.yaml

```diff
@@ -3632,61 +4295,180 @@ components:
       allOf:
-      - $ref: '#/components/schemas/BaseModel'
       - required:
-        - availabilityZone
         - name
-        - poolId
-        - profileId
         - size
         - tenantId
         type: object
         properties:
           tenantId:
             type: string
-            readOnly: true
+            example: e93b4c0934da416eb9c8d120c5d04d96
           userId:
             type: string
-            readOnly: true
+            example: e93b4c0934da416eb9c8d120c5d04d58
           name:
+            description: The name of the volume
+            minLength: 1
+            maxLength: 255
+            pattern: '[A-Za-z0-9_-]'
             type: string
+            example: volume1
           multiAttach:
             type: boolean
+            description: can this volume be attached to hosts
+            example: true
+            default: false
           description:
             type: string
+            description: This is description for volume
           size:
             type: integer
+            description: Requested size of volume in GB
             format: int64
-            example: 2
+            example: 20
+          availabilityZone:
+            type: string
+            description: The name of availability zone
+            example: zone1
+            default: default
+          profileId:
+            type: string
+            description: Id of profile, which describes storage capability policy
+            example: adace15c-bcaa-42e3-8aa7-500070c802bc
+            default: default_block
+          poolId:
+            type: string
+            description: The id of the pool, on which volume need to be created
+            example: cdefe15c-bcaa-42e3-8aa7-56070c802cd
+            default: opensds-volumes-default
+          snapshotId:
+            type: string
+            description: snapshotId is only required when user want to create volume
+              from specific snapshot.
+            example: bcace15c-bcaa-42e3-8aa7-400970c802cd
+          groupId:
+            type: string
+            description: The id of the group
+            example: bcace15c-bcaa-42e3-8aa7-400970c802er
+          snapshotFromCloud:
+            type: boolean
+            description: The snapshot from the cloud
+            example: true
+          replicationId:
+            type: string
+            description: replicationId is only required when user want to create volume
+              from specific replicated host.
+            example: bcace15c-bcaa-42e3-8aa7-400970c802cd
+          replicationDriverData:
+            type: object
+            additionalProperties:
+              type: string
+            example:
+              key1: value1
+              key2: value2
+          provisioningPolicy:
+            type: string
+            description: The provisioning policy of volume
+            example: thin
+            default: thick
+    VolumeRespSpec:
```

Comment:
Are the all parameters included in Volume Response Spec required?

Reply:
NO NO. As I had discussed to you,  currently analyzing how to show required parameter in response(a couple of ways to do this, either put * or optional word). we can adopt any one of the way.